### ISSUE TITLE
Calculation of projected graph edge 'weight' in network analysis is duplicative of 'length', may not be required

### ISSUE THREAD

**Issue by carlhiggs** [healthysustainablecities/global-indicators#243]
The introduction of an all pairs Dijkstra's shortest path function to support neighbourhood analysis in 2020 (https://github.com/global-healthy-liveable-cities/global-indicators/commit/b00df984256e4d445c1f4af65598e354ed85bbf2#), included the calculation of a `weight` attribute for edges.  This recast the existing `length` attribute associated with the projected network graph to an integer.  A comment suggested that the floating point value was recorded in a string, hence the recasting.  The sacrifice of precision, rounding to nearest meter, would have been intended to further speed up processing for larger cities.

However, re-approaching this step I can confirm that the current `length` attribute is a float data type; I'm also not convinced that the casting to integer at that point in the analysis is necessary or makes things quicker.

I believe these lines could be removed:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/b0c5fd80d0c4200d83925bf6725595ef3eded39e/process/subprocesses/_11_neighbourhood_analysis.py#L107-L117

while this line could be amended to use 'length' instead:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/b0c5fd80d0c4200d83925bf6725595ef3eded39e/process/subprocesses/_11_neighbourhood_analysis.py#L128-L130

**Comment by carlhiggs** [comment]
Just did a quick A/B comparison of the two approaches for a reasonably large-ish network (Baltimore), that suggests there isn't much benefit from the simplification --- took about the same amount of time for this operation (~3 minutes and 50 seconds) using either approach:

Using the 'weight' integer measure:
```
>>> all_pairs_d = pd.DataFrame(
...             [
...                 (k, v.keys())
...                 for k, v in tqdm(
...                     nx.all_pairs_dijkstra_path_length(
...                         G_proj, neighbourhood_distance, 'weight',
...                     ),
...                     total=total_nodes,
...                     unit='nodes',
...                     desc=' ' * 18,
...                 )
...             ],
...             columns=['osmid', 'nodes'],
...         ).set_index('osmid')
                  : 78038nodes [03:47, 343.75nodes/s]
```

Using the orgiinal 'length' float measure:
```
>>> all_pairs_d = pd.DataFrame(
...             [
...                 (k, v.keys())
...                 for k, v in tqdm(
...                     nx.all_pairs_dijkstra_path_length(
...                         G_proj, neighbourhood_distance, 'length',
...                     ),
...                     total=total_nodes,
...                     unit='nodes',
...                     desc=' ' * 18,
...                 )
...             ],
...             columns=['osmid', 'nodes'],
...         ).set_index('osmid')
                  : 78038nodes [03:49, 340.71nodes/s]
```

For a particular node (49399486), the respective results from these were:

using weight:
```
>>> grid.loc[gdf_nodes.loc[all_pairs_d.loc[49399486].nodes, 'grid_id'].dropna().unique()].mean()
<stdin>:1: FutureWarning: The default value of numeric_only in GeoDataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
pop_est                     70.318841
area_sqkm                    0.016533
pop_per_sqkm              4253.351317
intersection_count           4.239130
intersections_per_sqkm     256.410812
dtype: float64
```

using length:
```
>>> grid.loc[gdf_nodes.loc[all_pairs_d.loc[49399486].nodes, 'grid_id'].dropna().unique()].mean()
<stdin>:1: FutureWarning: The default value of numeric_only in GeoDataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
pop_est                     69.007407
area_sqkm                    0.016533
pop_per_sqkm              4174.027114
intersection_count           4.237037
intersections_per_sqkm     256.284189
dtype: float64
```

So - no important difference in results (that was rationale for using the integer simplification, under assumption it improved analysis performance presumably) or time to process (in which case, there is no need to simplify to integer).

**Comment by carlhiggs** [comment]
just ran the analysis with the above returning to using 'length' as cost (floating point) rather than 'weight' (integer, derived from length) for Las Palmas, with the following marginal results that indicate in addition to no practical difference in analysis time, there is no practical difference in results (and if there were, its a problem with the weight variable anyway).   

tldr, there's no harm but also effectively no benefit to using the integer-simplified weight variable when evaluating neighbourhood distance to justify the additional code complexity for doing that

|                                                   | using weight (integer) | using length(float; from which 'weight' was   derived) | Absolute difference (length-weight) | Relative difference (length-weight)/weight |
|---------------------------------------------------|------------------|--------------------------------------------------|-------------------------------------|--------------------------------------------|
| Area (sqkm)                                       | 103.3            | 103.3                                            | 0.000                               | 0.0000                                     |
| Population estimate                               | 366501           | 366501                                           | 0.000                               | 0.0000                                     |
| Population per sqkm                               | 3547             | 3547                                             | 0.000                               | 0.0000                                     |
| Intersections                                     | 6629             | 6629                                             | 0.000                               | 0.0000                                     |
| Intersections per sqkm                            | 64.2             | 64.2                                             | 0.000                               | 0.0000                                     |
| urban_sample_point_count                          | 37526            | 37526                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_fresh_food_market_score       | 48.81            | 48.81                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_convenience_score             | 51.02            | 51.02                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_pt_osm_any_score              | 82.91            | 82.91                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_public_open_space_any_score   | 70.43            | 70.43                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_public_open_space_large_score | 43.67            | 43.67                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_pt_gtfs_any_score             | 78.54            | 78.54                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_pt_gtfs_freq_30_score         | 73.46            | 73.46                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_pt_gtfs_freq_20_score         | 65.86            | 65.86                                            | 0.000                               | 0.0000                                     |
| pop_pct_access_500m_pt_any_score                  | 90.07            | 90.07                                            | 0.000                               | 0.0000                                     |
| pop_nh_pop_density                                | 9354.00          | 9362.18                                          | 8.188                               | 0.0009                                     |
| pop_nh_intersection_density                       | 176.13           | 176.17                                           | 0.042                               | 0.0002                                     |
| pop_daily_living                                  | 1.90             | 1.90                                             | 0.000                               | 0.0000                                     |
| pop_walkability                                   | 0.65             | 0.65                                             | 0.002                               | 0.0024                                     |
| local_nh_population_density                       | 7089.62          | 7088.12                                          | -1.508                              | -0.0002                                    |
| local_nh_intersection_density                     | 153.58           | 153.53                                           | -0.044                              | -0.0003                                    |
| local_daily_living                                | 1.43             | 1.43                                             | 0.000                               | 0.0000                                     |
| local_walkability                                 | -0.81            | -0.81                                            | 0.000                               | 0.0003                                     |
